index.js

```diff
@@ -1,21 +1,15 @@
 var loaderUtils = require('loader-utils');
 var MessageFormat = require('messageformat');
 
-module.exports = function(content) {
-  var query = loaderUtils.parseQuery(this.query);
-  var locale = query.locale || 'en';
-  var messages = (Array.isArray(this.inputValue) && typeof this.inputValue[0] === 'object') ? this.inputValue[0] : this.exec(content, this.resource);
```

Comment:
Ah, hadn't realised that about the JSON loaders -- only tested it without them on WebPack v2, where they're not required. Also FYI, [`this.exec`](https://webpack.js.org/api/loaders/#this-exec) is now deprecated.

Smart thing would probably be to look at the first non-white-space character of the input string, and consider it to be JSON only if it's `[` or `{`. If it's not, then exec it to read the object.